### ISSUE TITLE
linter: incorrect trigger of the `unicorn(no-array-fill-with-reference-type)` rule

### ISSUE THREAD

**Issue by niusia-ua** [oxc-project/oxc#23703]
### What version of Oxlint are you using?

1.70.0

### What command did you run?

oxlint --format stylish --deny-warnings

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "categories": {
    "correctness": "error",
    "suspicious": "error",
    "pedantic": "warn",
    "perf": "warn",
    "style": "off",
    "restriction": "off",
    "nursery": "off"
  },
  "plugins": ["oxc", "eslint", "typescript", "unicorn", "promise", "vue"],
}
```

### What happened?

The `unicorn(no-array-fill-with-reference-type)` rule incorrectly trigger for the `.fill()` method. For example, in my case, it is triggered for the `Graphics` operations in Pixi.js-related code:

```ts
const background = new Graphics()
  .rect(0, -this.#stages.horizontal.height, this.#width, this.#stages.horizontal.height)
  .fill({ color: 0x000000, alpha: 0.75 }); // Lint error.
this.#stages.horizontal.addChildAt(background, 0);
```

**Comment by baevm** [comment]
upstream rule also reports this case. Maybe this rule should be moved from the suspicious to the pedantic category, since it can have false positives

**Comment by camc314** [comment]
Hi! Thanks for reporting this issue.

This behaviour is aligned with [unicorn/no-array-fill-reference-type](https://github.com/sindresorhus/eslint-plugin-
unicorn/blob/main/docs/rules/no-array-fill-with-reference-type.md), so fixing this is not planned at present

That said, we can definitely try to harden this rule like some of the other unicorn rules, by trying to guess if it definitiely is/is not an array, but addressing that will be a lower priority. 

I would suggest disabling the rule in your project, that is using the `Pixi.js` code, as there will be lots of false positives.

Thanks!
